### ISSUE TITLE
`apropos` with two arguments segfaults

### ISSUE THREAD

**Issue by KeenS** [drmeister/clasp#52]
```
> (apropos "list-all")
COMPILER::LAMBDA-LIST-ALLOW-OTHER-KEYS
CORE::READ-LIST-ALLOW-CONSING-DOT
LIST-ALL-PACKAGES  Function
READER-LIST-ALLOW-CONSING-DOT  Function
> (apropos "list-all" :cmp)
zsh: segmentation fault (core dumped)  /usr/local/clasp/bin/clasp_mps_o
> (find-package :cmp)

#<COMPILER>
> (apropos "list-all" (find-package :cmp))
zsh: segmentation fault (core dumped)  /usr/local/clasp/bin/clasp_mps_o
```

Since documents are missing, apropos is a key feature.

**Comment by Shinmera** [comment]
Cannot confirm on b05884f due to a different issue:
```
> (apropos "list-all")
LIST-ALL-PACKAGES  Function
CORE::READ-LIST-ALLOW-CONSING-DOT
READER-LIST-ALLOW-CONSING-DOT  Function
> (apropos "list-all" :cmp)

```
at which point Clasp seems to do... something? forever. I can interrupt with C-c and invoke the restart back to the toplevel and it resumes fine though.

**Comment by drmeister** [comment]
Fixed problem by giving apropos-list-inner an extra argument that keeps track of what packages were visited

**Comment by Shinmera** [comment]
Fixed on 482d6ca5bd8ad30ff8c9b6a835989f3aec395e1a
```
> (apropos "list-all")
LIST-ALL-PACKAGES  Function
CORE::READ-LIST-ALLOW-CONSING-DOT
READER-LIST-ALLOW-CONSING-DOT  Function
> (apropos "list-all" :cmp)
LIST-ALL-PACKAGES  Function
CORE::READ-LIST-ALLOW-CONSING-DOT
READER-LIST-ALLOW-CONSING-DOT  Function
> (apropos "find")
FIND  Function
FIND-ALL-SYMBOLS  Function
FIND-BLOCK-NAMED-ENVIRONMENT  Function
CORE::FIND-BUILT-IN-TAG  Function
FIND-CLASS  Function
CORE::FIND-DECLARATIONS  Function
CORE::FIND-DIRECTIVE  Function
CORE::FIND-DOCUMENTATION  Function
CORE::FIND-DUPLICATES  Function
FIND-FILE-IN-LISP-PATH  Function
LLVM-SYS:FIND-FUNCTION-NAMED  Function
FIND-IF  Function
FIND-IF-NOT  Function
CLEAVIR-AST-TRANSFORMATIONS::FIND-LOAD-TIME-VALUE-ASTS  Function
CLASP-CLEAVIR-AST::FIND-LOAD-TIME-VALUE-ASTS  Function
FIND-METHOD  Function
CLOS:FIND-METHOD-COMBINATION  Function
CLEAVIR-AST-TO-HIR::FIND-OR-CREATE-LOCATION  Function
CORE::FIND-OR-MAKE-SYMBOL  Function
FIND-PACKAGE  Function
CORE::FIND-REGISTERED-TAG  Function
CORE::FIND-RELATIVE-PACKAGE  Function
FIND-RESTART  Function
CORE::FIND-RESTART-NEVER-FAIL  Function
FIND-SINGLE-DISPATCH-GENERIC-FUNCTION  Function
CLOS::FIND-SLOT-DEFINITION  Function
CORE::FIND-SUBCLASSES-OF-TYPE  Function
FIND-SYMBOL  Function
FIND-TAGBODY-TAG-ENVIRONMENT  Function
CORE::FIND-TYPE-BOUNDS  Function
FIND-UNWINDABLE-ENVIRONMENT  Function
CLASP-CLEAVIR::FIND-UNWINDS  Function
MAP-FIND  Function
AST-TOOLING:MATCH-FINDER
AST-TOOLING:NEW-MATCH-FINDER  Function
SETF-FIND-CLASS  Function
SETF-FIND-SINGLE-DISPATCH-GENERIC-FUNCTION  Function
STRING-FIND  Function
WALK-TO-FIND-SOURCE-INFO  Function
WALK-TO-FIND-SOURCE-POS-INFO  Function
> (defpackage #:foo (:use))

#<FOO>
> (apropos "find" '#:foo)

```
